src/conf.rs

```diff
@@ -0,0 +1,212 @@
+    #[serde(default = "info_ratelimit_default")]
+    info: RatelimitData,
+    #[serde(default = "message_create_ratelimit_default")]
+    message_create: RatelimitData,
+}
+
+impl Default for OprishRatelimits {
+    fn default() -> Self {
+        Self {
+            info: info_ratelimit_default(),
+            message_create: message_create_ratelimit_default(),
+        }
+    }
+}
+
+fn info_ratelimit_default() -> RatelimitData {
+    RatelimitData {
+        reset_after: 5,
+        limit: 2,
+    }
+}
+
+fn message_create_ratelimit_default() -> RatelimitData {
+    RatelimitData {
+        reset_after: 5,
+        limit: 10,
+    }
+}
+
+/// Pandemonium config.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct PandemoniumConf {
+    #[serde(default = "pandemonium_ratelimit_default")]
+    pub ratelimit: RatelimitData,
+}
+
+impl Default for PandemoniumConf {
+    fn default() -> Self {
+        Self {
+            ratelimit: pandemonium_ratelimit_default(),
+        }
+    }
+}
+
+fn pandemonium_ratelimit_default() -> RatelimitData {
+    RatelimitData {
+        reset_after: 10,
+        limit: 5,
+    }
+}
+
+/// Effis config.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct EffisConf {
+    pub file_size: String,
+    #[serde(default)]
+    pub ratelimit: EffisRatelimitData,
+}
+
+impl Default for EffisConf {
+    fn default() -> Self {
+        Self {
+            file_size: file_size_default(),
+            ratelimit: EffisRatelimitData::default(),
+        }
+    }
+}
+
+fn file_size_default() -> String {
+    "100MB".to_string()
+}
+
+/// Ratelimit config data.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct RatelimitData {
+    pub reset_after: u32,
+    pub limit: u32,
+}
+
+/// Effis ratelimit data config.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct EffisRatelimitData {
+    pub reset_after: u32,
+    pub limit: u32,
+    pub file_size_limit: String,
+}
+
+impl Default for EffisRatelimitData {
+    fn default() -> Self {
+        Self {
+            reset_after: 10,
+            limit: 25,
+            file_size_limit: "200MB".to_string(),
+        }
+    }
+}
+
+impl Conf {
+    /// Create a new [`Conf`].
+    ///
+    /// # Panics
+    ///
+    /// This function is *intended* to panic if a suitable config is not found.
+    ///
+    /// That also includes the config file's data failing to deserialise.
+    pub fn new(path: &str) -> Conf {
+        let data = fs::read_to_string(path).unwrap();
+        toml::from_str(&data).unwrap()
+    }
+
+    /// Create a new [`Conf`] with default config from the provided instance name.
+    pub fn from_name(instance_name: String) -> Self {
+        Self {
+            instance_name,
+            oprish: OprishConf::default(),
+            pandemonium: PandemoniumConf::default(),
+            effis: EffisConf::default(),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::conf::*;
+
+    #[test]
+    fn try_deserialize() {
+        let conf_str = r#"
```

Comment:
Could this be a global constant so the indentation doesn't look so weird?

Reply:
the point of it is it being in the cfg'd testing block which itself has one layer of indentation, it's actually not all that uncommon in rust as proven by [sqlx](https://github.com/launchbadge/sqlx#compile-time-verification)